### PR TITLE
Parsers remove folds

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: src/Streamly/Internal/Data/Parser/ParserD.hs [hunk 1]
```diff
@@ -377,7 +349,7 @@ takeGE cnt (Fold fstep finitial fextract) = Parser step initial extract
                       FL.Done _ ->
                         Error
                             $ "takeGE: the collecting fold terminated after "
-                                ++ "consuming " ++ show i1 ++ " elements"
+                                ++ "consuming" ++ show i1 ++ " elements"
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Parser/ParserD.hs:352]: Why remove the space?

File: src/Streamly/Internal/Data/Parser/ParserD.hs [hunk 2]
```diff
@@ -709,6 +659,7 @@ manyTill (Fold fstep finitial fextract)
             Error _ -> do
                 rR <- initialL
                 return $ Continue (cnt + 1) (ManyTillL 0 fs rR)
+    -- XXX the cnt is being used only by the assert
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Parser/ParserD.hs:662]: You can remove `cnt`. I think this remained when I removed out `Partial1` and `Done1`.

File: src/Streamly/Internal/Data/Parser/ParserD/Types.hs [hunk 3]
```diff
@@ -505,6 +499,7 @@ splitSome (Fold fstep finitial fextract) (Parser step1 initial1 extract1) =
     {-# INLINE step #-}
     step (Tuple3' st cnt (Left fs)) a = do
         r <- step1 st a
+        -- In the Left state, count is used only for the assert
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Parser/ParserD/Types.hs:502]: You can remove this. This remained when I removed `Partial1` and `Done1`.

**Review by adithyaov — APPROVED** [review]
Please address the comments.

File: src/Streamly/Internal/Data/Parser/ParserD.hs (hunk 1 above)
Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Parser/ParserD.hs:352]: That's by mistake, some issue during transferring the changes from one branch to another.

File: src/Streamly/Internal/Data/Parser/ParserD.hs (hunk 2 above)
Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Parser/ParserD.hs:662]: Let's keep the assert for now, its good for debugging.

File: src/Streamly/Internal/Data/Parser/ParserD/Types.hs (hunk 3 above)
Comment (harendra-kumar) [reply src/Streamly/Internal/Data/Parser/ParserD/Types.hs:502]: Let's keep the assert for now, its good for debugging.

**Pull request merged**